packages/experience-builder-sdk/src/hooks/useBreakpoints.ts

```diff
@@ -22,21 +22,23 @@ import { useCallback, useEffect, useState } from 'react';
  * and then decending by screen width. For mobile-first designs, the order would be ascending
  */
 export const useBreakpoints = (breakpoints: Breakpoint[]) => {
-  const [mediaQueryMatchers, initialMediaQueryMatches] = mediaQueryMatcher(breakpoints);
-
+  const fallbackBreakpointIndex = getFallbackBreakpointIndex(breakpoints);
+  const [mediaQueryMatchers, currentQueryMatches] = mediaQueryMatcher(breakpoints);
   const [mediaQueryMatches, setMediaQueryMatches] =
-    useState<Record<string, boolean>>(initialMediaQueryMatches);
+    useState<Record<string, boolean>>(currentQueryMatches);
 
-  const fallbackBreakpointIndex = getFallbackBreakpointIndex(breakpoints);
+  const activeBreakpointIndex = getActiveBreakpointIndex(
+    mediaQueryMatches,
+    fallbackBreakpointIndex
+  );
 
   // Register event listeners to update the media query states
   useEffect(() => {
-    const eventListeners = mediaQueryMatchers.map(({ id, signal }) => {
-      const onChange = () =>
-        setMediaQueryMatches((prev) => ({
-          ...prev,
-          [id]: signal.matches,
-        }));
+    const eventListeners = mediaQueryMatchers.map(({ signal }) => {
+      const onChange = () => {
+        const initialMediaQueryMatches = mediaQueryMatcher(breakpoints)[1];
```

Comment:
What is meant by "initial" here? As far as I see, this value is regenerate on every change, so it's the "current" value I guess.

Therefore, it might make more sense to rename `currentQueryMatches` in line 26 back to `initialQueryMatches`. it is recalculated on every render but we only use it as the initial value for the state value.